### PR TITLE
Update String»split examples w/ empty string & empty separator

### PR DESCRIPTION
**Pull request by kmbenjel**
#### Summary

Description was slightly different from the example in caption, I suggested a fix for that.

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ x] Fixes a typo, bug, or other error

### CODE REVIEW

**Review by sideshowbarker — COMMENTED** [review]
This change appears to be incorrect. Test case:
```js
const myArray = "".split("");
console.log(myArray);
// Array []
console.log(myArray.length);
// 0
```
That is, if the string and separator are both empty (`"".split("")`), an empty array is returned (not an array containing an empty string).

**Comment by sideshowbarker** [comment]
> Thank you for the follow-up The mistake is in the caption then, where you find: [""]

If by _caption_ you mean the comment in the code example at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split#using_split, then that comment is in fact correct as written — because that code  isn’t showing _“if the string and separator are both empty”_; instead it’s showing _“When the string is empty, `split()` returns an array containing one empty string, rather than an empty array”_.

But I see now that statement as written is ambiguous: It should instead say something more like, _When the string is empty **and no separator is specified**, `split()` returns an array containing one empty string, rather than an empty array_.

Would you be interested in updating this PR to add that wording?

And then maybe also update the PR to show an example with `"".split("")` — that is, the _“if the string and separator are both empty”_ case.

If re-opened the PR for now, so you can make those changes to the patch in the PR if you want.

**Comment by kmbenjel** [comment]
Ok.

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 1]
```diff
@@ -94,17 +94,22 @@ separator.
...
-const myString = ''
-const splits = myString.split()
+const emptyString = ''
+let splits = emptyString.split()
 
 console.log(splits)
-
 // ↪ [""]
+
+splits = emptyString.split(emptyString) // reassign `splits` variable
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:108]:
```suggestion
splits = emptyString.split(emptyString) // reassigns 'splits' variable
```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 2]
```diff
@@ -94,17 +94,22 @@ separator.
...
-When the string is empty, `split()` returns an array containing one empty
+When the string is empty and no separator is specified, `split()` returns an array containing one empty
 string, rather than an empty array. If the string and separator are both empty
 strings, an empty array is returned.
 
 ```js
-const myString = ''
-const splits = myString.split()
+const emptyString = ''
+let splits = emptyString.split()
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:103]:
```suggestion
```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 3]
```diff
@@ -94,17 +94,22 @@ separator.
...
 
 console.log(splits)
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:105]:
```suggestion
console.log(emptyString.split())
```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md (hunk 1 above)
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:108]:
```suggestion
```
```suggestion
splits = emptyString.split(emptyString) // reassign `splits` variable
```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 4]
```diff
@@ -94,17 +94,22 @@ separator.
...
-
 // ↪ [""]
+
+splits = emptyString.split(emptyString) // reassign `splits` variable
+
+console.log(splits)
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:110]:
```suggestion
console.log(emptyString.split(emptyString))
```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 5]
```diff
@@ -94,17 +94,20 @@ separator.
...
-const myString = ''
-const splits = myString.split()
-
-console.log(splits)
+const emptyString = ''
 
+console.log(emptyString.split()) // string is empty and no separator is specified
 // ↪ [""]
+
+
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:107]:
```suggestion
```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 6]
```diff
@@ -94,17 +94,19 @@ separator.
...
-
-console.log(splits)
+const emptyString = ''
 
+console.log(emptyString.split()) // string is empty and no separator is specified
 // ↪ [""]
+
+console.log(emptyString.split(emptyString)) // string and separator are both empty strings
+// ↪ []
+
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:109]:
```suggestion
```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 7]
```diff
@@ -94,17 +94,18 @@ separator.
...
-const splits = myString.split()
-
-console.log(splits)
+const emptyString = ''
 
+console.log(emptyString.split()) // string is empty and no separator is specified
 // ↪ [""]
+
+console.log(emptyString.split(emptyString)) // string and separator are both empty strings
+// ↪ []
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:108]:
```suggestion
// []
```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 8]
```diff
@@ -94,17 +94,18 @@ separator.
...
 string, rather than an empty array. If the string and separator are both empty
 strings, an empty array is returned.
 
 ```js
-const myString = ''
-const splits = myString.split()
+const emptyString = ''
 
-console.log(splits)
+console.log(emptyString.split()) // string is empty and no separator is specified
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:104]:
```suggestion
// string is empty and no separator is specified
console.log(emptyString.split())

```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 9]
```diff
@@ -94,17 +94,18 @@ separator.
...
+// [""]
 
-// ↪ [""]
+console.log(emptyString.split(emptyString)) // string and separator are both empty strings
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:107]:
```suggestion
// string and separator are both empty strings
console.log(emptyString.split(emptyString))
```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 10]
```diff
@@ -94,17 +94,21 @@ separator.
...
 ```js
-const myString = ''
-const splits = myString.split()
+const emptyString = ''
 
-console.log(splits)
+// string is empty and no separator is specified
+console.log(emptyString.split())
+
+// [""]
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:107]:
```suggestion
// [""]

```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 11]
```diff
@@ -94,17 +94,21 @@ separator.
...
+When the string is empty and no separator is specified, `split()` returns an array containing one empty
 string, rather than an empty array. If the string and separator are both empty
 strings, an empty array is returned.
 
 ```js
-const myString = ''
-const splits = myString.split()
+const emptyString = ''
+
+// string is empty and no separator is specified
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:104]:
```suggestion
// string is empty and no separator is specified
```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 12]
```diff
@@ -94,17 +94,20 @@ separator.
...
 
-When the string is empty, `split()` returns an array containing one empty
+When the string is empty and no separator is specified, `split()` returns an array containing one empty
 string, rather than an empty array. If the string and separator are both empty
 strings, an empty array is returned.
 
 ```js
-const myString = ''
-const splits = myString.split()
+const emptyString = ''
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:102]:
```suggestion
const emptyString = '';

```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 13]
```diff
@@ -94,17 +94,20 @@ separator.
...
+// string is empty and no separator is specified
+console.log(emptyString.split())
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:104]:
```suggestion
console.log(emptyString.split());
```

File: files/en-us/web/javascript/reference/global_objects/string/split/index.md [hunk 14]
```diff
@@ -94,17 +94,20 @@ separator.
...
+// [""]
 
-console.log(splits)
 
-// ↪ [""]
+// string and separator are both empty strings
+console.log(emptyString.split(emptyString))
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/global_objects/string/split/index.md:109]:
```suggestion
console.log(emptyString.split(emptyString));
```

**Review by sideshowbarker — APPROVED** [review]

**Comment by sideshowbarker** [comment]
Khalid, thanks much, and congrats on landing your first docs change here — welcome aboard 🎉

**Comment by kmbenjel** [comment]
Thank you very much! 😊

**Pull request merged**